pkg/controller/workload/workload_processor.go

```diff
@@ -272,54 +278,48 @@ func (p *Processor) storeEndpointWithService(sk *bpf.ServiceKey, sv *bpf.Service
 		log.Errorf("Update ServiceUpdate map failed, err:%s", err)
 		return err
 	}
-
-	p.WorkloadCache.UpdateRelationShip(ev.BackendUid, ek.ServiceId, ek.BackendIndex)
 	return nil
 }
 
-func (p *Processor) storeServiceEndpoint(workload_uid string, serviceName string) {
+// storeServiceEndpoint is to cache workload when the service has not been received
+func (p *Processor) storeServiceEndpoint(uid string, serviceName string) {
 	wls, ok := p.endpointsByService[serviceName]
 	if !ok {
 		p.endpointsByService[serviceName] = make(map[string]struct{})
 		wls = p.endpointsByService[serviceName]
 	}
 
-	wls[workload_uid] = struct{}{}
+	wls[uid] = struct{}{}
 }
 
-func (p *Processor) deleteResidualServicesWithWorkload(workload *workloadapi.Workload, services []string) error {
-	var (
-		err       error
-		serviceId uint32
-	)
-
-	if services == nil {
+// handleWorkloadUnboundServices handles when a workload's belonging services removed
+func (p *Processor) handleWorkloadUnboundServices(workload *workloadapi.Workload, services []string) error {
+	if len(services) == 0 {
 		return nil
 	}
 
-	log.Infof("deleteResidualServicesWithWorkload: %v", services)
-	eks := make([]bpf.EndpointKey, 0)
-	workloadUid := p.hashName.StrToNum(workload.GetUid())
-	serviceIds := make(map[uint32]struct{})
+	log.Infof("handleWorkloadUnboundServices %s: %v", workload.Uid, services)
```

Comment:
Can replace workload.Uid with workload.name?

Reply:
isn't Uid contains more info? like namespace and kind info